File: python-sdk/src/boxsdk/models/shared/watermark.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from boxsdk import utils


@dataclass_json
@dataclasses.dataclass
class WatermarkWatermark:
    created_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('created_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    modified_at: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('modified_at'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    

@dataclass_json
@dataclasses.dataclass
class Watermark:
    r"""Watermark
    A watermark is a semi-transparent overlay on an embedded file
    preview that displays a viewer's email address or user ID
    and the time of access over a file's content
    """
    
    watermark: Optional[WatermarkWatermark] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('watermark') }})
    

File: sdks/java/lib/src/main/java/models/shared/Watermark.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Watermark
 * A watermark is a semi-transparent overlay on an embedded file
 * preview that displays a viewer's email address or user ID
 * and the time of access over a file's content
**/
public class Watermark {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("watermark")
    public WatermarkWatermark watermark;
    public Watermark withWatermark(WatermarkWatermark watermark) {
        this.watermark = watermark;
        return this;
    }
}